script.js

```diff
@@ -1,406 +1,501 @@
+		tempN = lastN[lastN.length - 1];
+		tempArr.push(tempN);
+	}
+	booksLastName = books;
+	for (i = 0; i < books.length; i++) {
+		booksLastName[i].lastname = tempArr[i];
+	}
+	sortLast(booksLastName);
+}
+
+function sortLast(booksLastName) {
+	//clear existing content;
+	main.innerHTML = "";
+	// make new array with sorted results on last name
+	const booksByAuthorLastname = booksLastName.slice().sort((a, b) => a.lastname.localeCompare(b.lastname));
+	// pass new array into function for filtering books.
+	displayFilteredBooks(booksByAuthorLastname);
+}
+
+//----------- MAKE THE BUTTONS -----------//
+
+
+//genre
+genres.forEach((genre) => {
+	const button = document.createElement("button");
+	button.classList.add('button');
+	button.textContent = genre;
+	button.addEventListener("click", () => filterByGenre(genre));
+	genreBtnContainer.appendChild(button);
+});
+
+//rating
+rating.forEach((rating) => {
+	const button = document.createElement("button");
+	button.classList.add('button', 'sort');
+	button.textContent = rating;
+	button.addEventListener("click", () => filterBooksbyRating(rating));
+	sortBtnContainer.appendChild(button);
+});
+
+//sorting on author and title
+const booksByTitle = books.slice().sort((a, b) => a.title.localeCompare(b.title));
+const sortButton = document.createElement("button");
+sortButton.classList.add('button');
+sortButton.textContent = "Title";
+nameBtnContainer.appendChild(sortButton);
+
+
+const booksByAuthor = books.slice().sort((a, b) => a.author.localeCompare(b.author));
+const sortAuthorButton = document.createElement("button");
+sortAuthorButton.classList.add('button');
+sortAuthorButton.textContent = "Firstname";
+nameBtnContainer.appendChild(sortAuthorButton);
+
+//sorting on last name
+const sortAuthorLastNameButton = document.createElement("button");
+sortAuthorLastNameButton.classList.add('button');
+sortAuthorLastNameButton.textContent = "Lastname";
+nameBtnContainer.appendChild(sortAuthorLastNameButton);
+
+//sort on highest rating
+const sortRatingHighButton = document.createElement("button");
+sortRatingHighButton.classList.add('button', 'sort');
+sortRatingHighButton.textContent = "Decending";
+sortBtnContainer.appendChild(sortRatingHighButton);
+
+//lowest rating
+const sortRatingLowBtn = document.createElement("button");
+sortRatingLowBtn.classList.add('button', 'sort');
+sortRatingLowBtn.textContent = "Acending";
+sortBtnContainer.appendChild(sortRatingLowBtn);
+
+//random
+const randomButton = document.createElement("button");
+randomButton.classList.add('button');
+randomButton.textContent = "Random";
+buttonContainer.appendChild(randomButton);
+
+//---------- FILTER & SORT FUNCTIONS -------------//
+
+//function for sort high to low rating
+function sortHigh(a, b) {
+	main.innerHTML = ""; //clear existing content;
+	const booksByHighRating = books.sort((a, b) => b.rating - a.rating);
+	displayFilteredBooks(booksByHighRating);
+}
+
+/*
+function getRandom() {
```

Comment:
maybe you can take this out

Reply:
take what out? the function?

Reply:
ah!  yes - I forgot to remove it - thank you :)